packages/flutter/test/material/app_bar_theme_test.dart

```diff
@@ -1199,13 +1223,110 @@ void main() {
     // the web and the rest of Flutter's target platforms.
   }, skip: kIsWeb); // https://github.com/flutter/flutter/issues/87364
 
+  testWidgets('Local AppBarTheme overrides defaults', (WidgetTester tester) async {
```

Comment:
Can we also add a test to check local AppBarTheme can overrides global AppBarTheme?